ui/src/components/PulseBindings/index.jsx

```diff
@@ -0,0 +1,136 @@
+import { hot } from 'react-hot-loader';
+import React, { Component, Fragment } from 'react';
+import { withApollo } from 'react-apollo';
+import classNames from 'classnames';
+import { withStyles } from '@material-ui/core/styles';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import ListSubheader from '@material-ui/core/ListSubheader';
+import Tooltip from '@material-ui/core/Tooltip';
+import TextField from '@material-ui/core/TextField';
+import Typography from '@material-ui/core/Typography';
+import IconButton from '@material-ui/core/IconButton';
+import PlusIcon from 'mdi-react/PlusIcon';
+import DeleteIcon from 'mdi-react/DeleteIcon';
+
+@hot(module)
+@withApollo
+@withStyles(theme => ({
+  iconButton: {
+    '& svg': {
+      fill: theme.palette.text.primary,
+    },
+  },
+  plusIcon: {
+    marginTop: 80,
+  },
+  deleteIcon: {
+    marginRight: -15,
+    [theme.breakpoints.down('sm')]: {
+      marginRight: -14,
+    },
+  },
+  inputWrapper: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  inputList: {
+    flex: 1,
+  },
+  bindingListItem: {
+    paddingTop: 0,
+    paddingBottom: 0,
+  },
+}))
+export default class PulseBindings extends Component {
+  render() {
+    const {
+      pulseExchange,
+      pattern,
+      bindings,
+      classes,
+      onBindingAdd,
+      onBindingRemove,
+      onChange,
+    } = this.props;
```

Comment:
Let's add `static propTypes`. We can also add `static defaultProps` if certain props are not required.

Reply:
I feel all the props are required for this.
I have written `static propTypes` something like this.

```
static propTypes = {
    pulseExchange: string.isRequired,
    pattern: string.isRequired,
    bindings: arrayOf(object).isRequired,
    onBindingAdd: func.isRequired,
    onBindingRemove: func.isRequired,
    onChange: func.isRequired,
  };
```

Also, I am not sure if we would not need `static defaultProps` in this.

Let me know if there is anything missing

Reply:
Looks good.